src/main/java/net/snowflake/ingest/streaming/internal/DataValidationUtil.java

```diff
@@ -552,44 +566,83 @@ static byte[] validateAndParseBinary(
    *
    * <ul>
    *   <li>String
-   *   <li>LocalTime
-   *   <li>OffsetTime
+   *   <li>{@link LocalTime}
+   *   <li>{@link OffsetTime}
    * </ul>
    */
   static BigInteger validateAndParseTime(String columnName, Object input, int scale) {
-    SFTimestamp timestamp;
-    if (input instanceof String) {
-      String stringInput = (String) input;
-      timestamp =
-          createDateTimeFormatter()
-              .parse(stringInput, GMT, 0, SnowflakeDateTimeFormat.TIME, true, null);
-    } else if (input instanceof LocalTime) {
-      timestamp =
-          timeStampFromLocalDateTime(((LocalTime) input).atDate(LocalDate.ofEpochDay(0)), GMT);
+    if (input instanceof LocalTime) {
+      LocalTime localTime = (LocalTime) input;
+      return BigInteger.valueOf(localTime.toNanoOfDay()).divide(Power10.sb16Table[9 - scale]);
     } else if (input instanceof OffsetTime) {
-      timestamp =
-          timeStampFromLocalDateTime(
-              ((OffsetTime) input).toLocalTime().atDate(LocalDate.ofEpochDay(0)), GMT);
-    } else {
-      throw typeNotAllowedException(
-          columnName, input.getClass(), "TIME", new String[] {"String", "LocalTime", "OffsetTime"});
-    }
+      return validateAndParseTime(columnName, ((OffsetTime) input).toLocalTime(), scale);
+    } else if (input instanceof String) {
+      String stringInput = (String) input;
+      {
+        // First, try to parse LocalTime
+        LocalTime localTime = catchParsingError(() -> LocalTime.parse(stringInput));
+        if (localTime != null) {
+          return validateAndParseTime(columnName, localTime, scale);
+        }
+      }
+
+      {
+        // Alternatively, try to parse OffsetTime
+        OffsetTime offsetTime = catchParsingError((() -> OffsetTime.parse(stringInput)));
+        if (offsetTime != null) {
+          return validateAndParseTime(columnName, offsetTime.toLocalTime(), scale);
+        }
+      }
+
+      {
+        // Alternatively, try to parse integer-stored time
+        Instant parsedInstant = catchParsingError(() -> parseInstantGuessScale(stringInput));
+        if (parsedInstant != null) {
+          return validateAndParseTime(
+              columnName,
+              LocalDateTime.ofInstant(parsedInstant, ZoneOffset.UTC).toLocalTime(),
+              scale);
+        }
+      }
 
-    if (timestamp == null) {
       throw valueFormatNotAllowedException(
           columnName,
           input,
           "TIME",
           "Not a valid time, see"
-              + " https://docs.snowflake.com/en/user-guide/date-time-input-output.html#time-formats"
+              + " https://docs.snowflake.com/en/LIMITEDACCESS/snowpipe-streaming.html"
```

Comment:
could you add a TODO comment to update this in the future?

Reply:
TODO added